PubNub/Core/PubNub+Core.m

```diff
@@ -570,7 +570,9 @@ - (void)prepareRequiredParameters {
         @"pnsdk":[NSString stringWithFormat:@"PubNub-%@%%2F%@", kPNClientName, kPNLibraryVersion]
     } mutableCopy];
 
-    if (self.configuration.authKey.length) {
+    if (self.configuration.authToken.length) {
+        queryComponents[@"auth"] = self.configuration.authToken;
```

Comment:
With other SDKs we reused a concept of the very small Token Manager. Maybe we should be consistent and not use configuration for that matter?

```
class TokenManager:
    def __init__(self):
        self.token = None

    def set_token(self, token):
        self.token = token

    def get_token(self):
        return self.token

    @staticmethod
    def unwrap_token(token):
        token = token.replace("_", "/").replace("-", "+")
        byte_array = base64.b64decode(token)
```

Reply:
@650elx Objective-C never had it and also it is pretty strange to have class solely for one value storage.

Reply:
IMO if you have some data + behaviour related to that data (parsing token etc.) it's worth encapsulating that into class.
IIRC the specification written by @raipubnub was build on old PAMv3 code with simplified TokenManager.

But! It's not a dealbreaker for me - it's up to your preference :)

Reply:
@650elx specs doesn’t state, that it should be simplified. It is said, that token manager can be removed or greatly simplified.

token class knows how to process CBOR and state it’s inner state from it - it is not manager responsibility.
Then all responsibility my of manager end up in value storage. Manager can’t be responsible for deciding on whether token or auth key should be used.